src/param_env.md

```diff
@@ -0,0 +1,17 @@
+# Parameter Environment
+
+When working with associated constants or generic types it is often relevant to
```

Comment:
This should be more general, maybe "generic and/or associated items (types, constants, functions/methods)"?